packages/react-data-grid/src/ReactDataGrid.tsx

```diff
@@ -219,194 +207,80 @@ const ReactDataGridBase = forwardRef(function ReactDataGrid<R extends {}>({
-            rowsSelected.push({ rowIdx: i, row: rowGetter(i) });
-          }
-
-          if (typeof rowSelection.onRowsSelected === 'function') {
-            rowSelection.onRowsSelected(rowsSelected);
-          }
-
-          handled = true;
-        } else if (rowIdx < lastRowIdxUiSelected) {
-          const rowsSelected = [];
-
-          for (let i = rowIdx; i <= lastRowIdxUiSelected - 1; i++) {
-            rowsSelected.push({ rowIdx: i, row: rowGetter(i) });
-          }
-
-          if (typeof rowSelection.onRowsSelected === 'function') {
-            rowSelection.onRowsSelected(rowsSelected);
-          }
-
-          handled = true;
-        }
-
-        if (handled) {
-          setLastRowIdxUiSelected(rowIdx);
-        }
-
-        return handled;
-      }
-
-      return false;
-    }
-
-    function handleNewRowSelect(rowIdx: number, rowData: R) {
-      const { current } = selectAllCheckboxRef;
-      if (current && current.checked === true) {
-        current.checked = false;
-      }
-
-      if (rowSelection) {
-        const { keys, indexes, isSelectedKey } = rowSelection.selectBy as { [key: string]: unknown };
-        const isPreviouslySelected = isRowSelected(keys, indexes, isSelectedKey, rowData, rowIdx);
-
-        setLastRowIdxUiSelected(isPreviouslySelected ? -1 : rowIdx);
-        const cb = isPreviouslySelected ? rowSelection.onRowsDeselected : rowSelection.onRowsSelected;
-        if (typeof cb === 'function') {
-          cb([{ rowIdx, row: rowData }]);
-        }
-      }
-    }
-
-    // columnKey not used here as this function will select the whole row,
-    // but needed to match the function signature in the CheckboxEditor
-    function handleRowSelect(rowIdx: number, columnKey: keyof R, rowData: R, event: React.ChangeEvent<HTMLInputElement>) {
-      event.stopPropagation();
-
-      if (canUseNewRowSelection()) {
-        if (rowSelection && rowSelection.enableShiftSelect === true) {
-          if (!handleShiftSelect(rowIdx)) {
-            handleNewRowSelect(rowIdx, rowData);
-          }
-        } else {
-          handleNewRowSelect(rowIdx, rowData);
-        }
-      } else { // Fallback to old onRowSelect handler
-        const newSelectedRows = enableRowSelect === 'single' ? [] : [...selectedRows];
-        const selectedRow = getSelectedRow(newSelectedRows, rowData[rowKey]);
-        if (selectedRow) {
-          selectedRow.isSelected = !selectedRow.isSelected;
-        } else {
-          (rowData as SelectedRow<R>).isSelected = true;
-          newSelectedRows.push(rowData as SelectedRow<R>);
-        }
-        setSelectedRows(newSelectedRows);
-        if (onRowSelect) {
-          onRowSelect(newSelectedRows.filter(r => r.isSelected === true));
-        }
+  const gridColumns = useMemo(() => {
+    // TODO: handle row selection without checkbox, i.e using row click
+    if (!rowSelection || !rowSelection.showCheckbox) return columns;
```

Comment:
Need to figure out how to optimize this block
- `rowSelection` is an object so it will always be different.
- it is possible that `rowGetter` and `onSelectedRowsChange` and created on every render so how should we handle it?
https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often

Reply:
- I don't like object props, we should just have `selectedRows?: Set` and `onSelectedRowsChange?(): void` and so on.
- is generating the `gridColumns` a heavy computation at all?

Reply:
1) Let's remove the object prop 
2) Probably not but changing columns may trigger other unnecessary updates